### PR TITLE
Add test for uploading invalid poms

### PR DESCRIPTION
**Pull request by tomasdavidorg**

### CODE REVIEW

File: guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/helpers/UploadInvalidPomTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        resolvePom("org/guvnor/m2repo/backend/server/helpers/broken-pom.xml");
+    }
+
+    @Test(expected = ProjectBuildingException.class)
+    public void testNonExistingParentGavInPom() throws Exception {
+        resolvePom("org/guvnor/m2repo/backend/server/helpers/non-existing-parent-gav-pom.xml");
+    }
+
+    private void resolvePom(String path) throws Exception {
+        InputStream pomInputStream = this.getClass().getClassLoader().getResourceAsStream(path);
```
Comment (manstis) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/helpers/UploadInvalidPomTest.java]: It'd be nice if you closed the ```InputStream``` after the test (or use "try with resources").

**Comment by manstis** [comment]
@tomasdavidorg Hi, thanks for the PR. Other than a small issue; it looks OK - was you planning on testing for any of the other ```Exceptions``` that can be thrown from ```PomModelResolver.parseXXX(...)``` methods; or is there a particular use-case that highlighted the need for these specific tests?

**Comment by tomasdavidorg** [comment]
Hi @manstis, this tests are moved from our selenium test suite in our effort to reduce it as much as possible. There is other special need for these specific tests.

Thanks for the comment. I hope now it is ok.

**Comment by manstis** [comment]
@tomasdavidorg Thanks for explaining the source of these tests; if QE are migrating more tests to Community to minimise your Selenium suite please comment the PR as I'll ask less questions ;)

**Pull request merged**